include/moqt/core/transport.h

```diff
@@ -0,0 +1,226 @@
+         */
+        void UnpublishTrack(TransportConnId conn_id, std::shared_ptr<PublishTrackHandler> track_delegate);
+
+        /**
+         * @brief Get the instance status
+         *
+         * @return Status indicating the state/status of the instance
+         */
+        Status GetStatus();
+
+        // --------------------------------------------------------------------------
+        // Metrics
+        // --------------------------------------------------------------------------
+
+        /**
+         * @brief Connection metrics for server accepted connections
+         *
+         * @details Connection metrics are updated real-time and transport quic metrics on
+         *      Config::metrics_sample_ms period
+         */
+        std::map<TransportConnId, ConnectionMetrics> connection_metrics_;
+
+      private:
+        // -------------------------------------------------------------------------------------------------
+        // Transport Delegate/callback functions
+        // -------------------------------------------------------------------------------------------------
+
+        void on_new_data_context([[maybe_unused]] const TransportConnId& conn_id,
+                                 [[maybe_unused]] const DataContextId& data_ctx_id) override
+        {
+        }
+
+        void on_connection_status(const TransportConnId& conn_id, const TransportStatus status) override;
+        void on_new_connection(const TransportConnId& conn_id, const TransportRemote& remote) override;
+        void on_recv_stream(const TransportConnId& conn_id,
+                            uint64_t stream_id,
+                            std::optional<DataContextId> data_ctx_id,
+                            const bool is_bidir = false) override;
+        void on_recv_dgram(const TransportConnId& conn_id, std::optional<DataContextId> data_ctx_id) override;
+
+        // -------------------------------------------------------------------------------------------------
+        // End of transport delegate/callback functions
+        // -------------------------------------------------------------------------------------------------
+
+        struct ConnectionContext
+        {
+            TransportConnId conn_id;
+            std::optional<uint64_t> ctrl_data_ctx_id;
+            bool setup_complete{ false }; ///< True if both client and server setup messages have completed
+            uint64_t client_version{ 0 };
+            std::optional<messages::MoqtMessageType>
+              ctrl_msg_type_received; ///< Indicates the current message type being read
+
+            uint64_t current_subscribe_id{ 0 }; ///< Connection specific ID for subscribe messages
+
+            /// Track namespace/name by received subscribe IDs
+            /// Used to map published tracks to subscribes in client mode
+            std::map<uint64_t, std::pair<uint64_t, uint64_t>> recv_sub_id;
```

Comment:
should this be a map of subscribe_id to FullTrackName here ?

Reply:
Again, this is not part of the public API as these are private members used by the implementation.

Reply:
I do understand this is not part of the api code to review but I did review since it was part of include/ which i feel still as public. Regardless, i was curious to understand if the comment makes sense in general ..